Competition2017/src/competition/subsystems/drive/commands/FieldOrientedTankDriveWithJoystick.java

```diff
@@ -0,0 +1,69 @@
+package competition.subsystems.drive.commands;
+
+import java.util.function.DoubleSupplier;
+
+import com.google.inject.Inject;
+
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import xbot.common.controls.sensors.XJoystick;
+import xbot.common.math.PIDFactory;
+import xbot.common.math.PIDManager;
+
+public class FieldOrientedTankDriveWithJoystick extends BaseDriveCommand {
+
+    private final PoseSubsystem poseSubsystem;
+    private final OperatorInterface oi;
+    
+    private XJoystick joystick;
+    
+    private final PIDManager headingDrivePid;
+    public final double defaultPValue = 1 / 80d;
+
+    protected DoubleSupplier targetHeadingSupplier;
+    
+    @Inject
+    public FieldOrientedTankDriveWithJoystick(
+            OperatorInterface oi, 
+            DriveSubsystem driveSubsystem, 
+            PoseSubsystem poseSubsystem,
+            PIDFactory pidFactory) {
+        super(driveSubsystem);
+        this.poseSubsystem = poseSubsystem;
+        this.oi = oi;
+        this.joystick = oi.leftJoystick;
+        headingDrivePid = pidFactory.createPIDManager("Rotate to joystick vector", defaultPValue, 0, 0, 0, 1, -1, 3, 3.0 / 20, 1);
+    }
+
+    public enum JoystickSide {
+        LeftJoystick, RightJoystick
```

Comment:
Do we want to structure it like this instead of passing an `XJoystick` directly? I did that approach on one of the vision-related commands... we should try to normalize it one way or another.